apps/portal/controllers/gadgets.jag

```diff
@@ -0,0 +1,99 @@
+<%
+/**
+ * Copyright 2016 WSO2, Inc. (http://wso2.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * All the requests that starts with /gadgets/ will be handled with this controller
+ */
+
+var fresh = false;
+isDefault = false;
+var bgColor;
+var pageId;
+(function () {
+    var log = new Log();
+    var matcher = new URIMatcher(request.getRequestURI());
+    var uriElements;
+    var dashboardId;
+    var page;
+    var isStandAlone = true; //To check whether the request is for stand alone gadget/ pub-sub gadget
+    var i18n = require("i18n");
+    var gadgets;
+    i18n.init(request, utils.getLocaleResourcePath());
+
+    if (true) {
```

Comment:
Initially, in order to check the functionality of this, we provided this feature without security, later this was removed with this PR #205